src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,29 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+
+	export namespace diagnostics {
+		/**
+		 * Diagnostics object for handling error codes for a connection provider.
+		 */
+		export interface ErrorDiagnostics {
+			/**
+			 * Takes an error code, connection profile, and error message and performs corrective tasks.
+			 * @param errorCode The error code of the connection error.
+			 * @param errorMessage The error message of the connection error.
+			 * @param connection The connection profile that caused the error.
+			 */
+			handleError(errorCode: number, errorMessage: string, connection: connection.ConnectionProfile): Thenable<boolean>;
```

Comment:
use Promise<boolean> as return type

Reply:
azdata APIs should always use Thenables (the idea is that extensions are free to use other promise libraries if they want).